File: shoring/fxapp/src/main/java/com/panopset/fxapp/FontSize.java
package com.panopset.fxapp;

import com.panopset.compat.Stringop;

public enum FontSize {
  SMALL(9, 1, 1.5), MEDIUM(12, 2, 2), LARGE(16, 3, 3), SUPER(24, 4, 4), CINEMA(48, 5, 5);
  private final int val;
  private final double imgRatio;
  private final double svgRatio;

  FontSize(int fontSize, double imgRatio, final double svgRatio) {
    val = fontSize;
    this.imgRatio = imgRatio;
    this.svgRatio = svgRatio;
  }

  public static FontSize findFromName(String fontName) {
    if (!Stringop.isPopulated(fontName)) {
      return DEFAULT_SIZE;
    }
    for (FontSize size : FontSize.values()) {
      if (size.name().equals(fontName)) {
        return size;
      }
    }
    return DEFAULT_SIZE;
  }

  public static FontSize findFromValue(String fontSzAsStr) {
    if (!Stringop.isPopulated(fontSzAsStr)) {
      return DEFAULT_SIZE;
    }
    for (FontSize size : FontSize.values()) {
      if (String.format("%d", size.val).equals(fontSzAsStr)) {
        return size;
      }
    }
    return DEFAULT_SIZE;
  }

  public static FontSize find(final int fontSize) {
    for (FontSize size : FontSize.values()) {
      if (size.getValue() == fontSize) {
        return size;
      }
    }
    for (FontSize size : FontSize.values()) {
      if (size.getValue() > fontSize) {
        return size;
      }
    }
    return DEFAULT_SIZE;
  }

  public int getValue() {
    return val;
  }

  public double getImgRatio() {
    return imgRatio;
  }

  public double getSvgRatio() {
    return svgRatio;
  }
  
  public static final FontSize DEFAULT_SIZE = FontSize.MEDIUM;
}

File: code/shoring/fxapp/src/main/kotlin/com/panopset/fxapp/FontSize.kt
package com.panopset.fxapp

import com.panopset.compat.Stringop

enum class FontSize(val value: Int, val imgRatio: Double, val svgRatio: Double) {
    SMALL(9, 1.0, 1.5),
    MEDIUM(12, 2.0, 2.0),
    LARGE(16, 3.0, 3.0),
    SUPER(24, 4.0, 4.0),
    CINEMA(48, 5.0, 5.0);

    companion object {
        fun findFromName(fontName: String): FontSize {
            if (!Stringop.isPopulated(fontName)) {
                return DEFAULT_SIZE
            }
            for (size in entries) {
                if (size.name == fontName) {
                    return size
                }
            }
            return DEFAULT_SIZE
        }

        fun findFromValue(fontSzAsStr: String): FontSize {
            if (!Stringop.isPopulated(fontSzAsStr)) {
                return DEFAULT_SIZE
            }
            for (size in entries) {
                if (String.format("%d", size.value) == fontSzAsStr) {
                    return size
                }
            }
            return DEFAULT_SIZE
        }

        fun find(fontSize: Int): FontSize {
            for (size in entries) {
                if (size.value == fontSize) {
                    return size
                }
            }
            for (size in entries) {
                if (size.value > fontSize) {
                    return size
                }
            }
            return DEFAULT_SIZE
        }

        val DEFAULT_SIZE = MEDIUM
    }
}
